### ISSUE TITLE
REQ: If a method's number of args changes, make sure that mapping.json is updated, and a new TestCase import is made

### ISSUE THREAD

**Issue by rarebreed** [RedHatQE/polarize#57]
I believe this is already handled if the annotation for the method has update=true, but I need to check.  This requirement came to light when a test was run, and the xunit report failed to get generated.  This is because the XUnitReporter.getPropertiesFromMethod() compares the number of arguments from the mapping.json file for a particular method, and the number of arguments from the IResult.getParameters() method call.  If the size is different, it throws an error.  This can be caused when a method is refactored and the number of parameters that the method takes changes.

For one, this kind of error could be detected at compile time, rather than fail at runtime.  When the annotation processor is running, have it look at the number of arguments via reflection for that method, and compare to how many are in the mapping.json file.

Secondly, the current code doesn't tell you what method had an incorrect number of arguments.  So change the debugging log info to get the name of the method that is failing.

Third, test the scenario where a method's number of arguments changes.  Make sure during compilation that the mapping.json gets the updated parameters, and also that an Import Request should be made.  The Import Request is required, because the way that iterations in Polarion works is that the number of elements of:

```xml
<property name="polarion-parameter-arg0" value="   stage_auto_testuser1"/>
```

Under at <testcase> must match how many are in the TestStep of the TestCase (which in the XML file is the number of <parameter /> elements) eg:

```xml
<test-step-column id="step">
    <parameter name="bugzilla" scope="local"/>
    <parameter name="bashCommand" scope="local"/>
    <parameter name="expectedCompletions" scope="local"/>
</test-step-column>
```

In this case, the XML has 3 parameter elements, so in Polarion, the TestSteps will have 3 args.  When the xunit is generated, the mapping.json file must also contain 3 elements:

```json
"RedHatEnterpriseLinux7" : {
      "id" : "RHEL7-68140",
      "parameters" : [ "bugzilla", "bashCommand", "expectedCompletions" ]
    }
```

**Comment by rarebreed** [comment]
So, it looks like if the update field is set to true, it will regenerate the XML, but not the mapping file.  So I need to correct this

**Comment by rarebreed** [comment]
Polarize will now do a couple of things:

- Compare the number of args in the method and what is in the mapping file for that method
  - If they are different and testcase importer is disabled, a new warning is printed during compilation
- Added to the audit file which methods need a TestCase import, if the testcase importer is disabled
- Regenerates the mapping file if update=true in annotation
- Throws an error during compile time if after map file regeneration, the param count is still not equal
- Added an option to specify an env var called POLARIZE_CONFIG to allow lein compile to point to a non-default config file location
- At runtime, provide more debug info if the param count of the method and map file still not the same (shouldn't happen, but there just in case)

**Comment by rarebreed** [comment]
Implemented in e733c07
